### PR TITLE
feat: demonstrate asking for post notification permission

### PR DESCRIPTION
**Pull request by kroikie**
Add snippets of requesting and handling the result of a POST_NOTIFICATION permission request.

### CODE REVIEW

File: messaging/app/src/main/java/com/google/firebase/example/messaging/MainActivity.java [hunk 1]
```diff
@@ -102,4 +109,41 @@ public void onComplete(@NonNull Task<String> task) {
...
+    @RequiresApi(33)
+    // [START ask_post_notifications]
+    private void askNotificationPermission() {
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.POST_NOTIFICATIONS) ==
+                PackageManager.PERMISSION_GRANTED) {
+            // FCM SDK (and your app) can post notifications.
+        } else if (shouldShowRequestPermissionRationale(Manifest.permission.POST_NOTIFICATIONS)) {
+            // TODO: display an educational UI explaining to the user the features that will be enabled
+            //       by them granting the POST_NOTIFICATION permission. This UI should provide the user
+            //       "OK" and "No thanks" buttons. If the user selects "OK" directly request the permission.
```
Comment (egilmorez) [inline messaging/app/src/main/java/com/google/firebase/example/messaging/MainActivity.java:122]: Suggest a couple commas to place:

"If the user selects "OK," directly request the permission.
            //       If the user selects "No thanks," allow

File: messaging/app/src/main/java/com/google/firebase/example/messaging/MainActivity.java [hunk 2]
```diff
@@ -102,4 +109,41 @@ public void onComplete(@NonNull Task<String> task) {
...
+                                           int[] grantResults) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+        switch (requestCode) {
+            case NOTIFICATION_REQUEST_CODE:
+                // If request is cancelled, the result arrays are empty.
+                if (grantResults.length > 0 &&
+                        grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    // FCM SDK (and your app) can post notifications.
+                } else {
+                    // TODO: Inform user that that your app will not show notifications
```
Comment (egilmorez) [inline messaging/app/src/main/java/com/google/firebase/example/messaging/MainActivity.java:143]: Suggest period after "notifications."

**Review by egilmorez — APPROVED** [review]
LG with nits!

I will crib some of the guidance in the comments for the guide page.

File: messaging/app/src/main/java/com/google/firebase/example/messaging/kotlin/MainActivity.kt [hunk 3]
```diff
@@ -90,4 +96,40 @@ class MainActivity : AppCompatActivity() {
...
+    @RequiresApi(33)
+    // [START ask_post_notifications]
+    private fun askNotificationPermission() {
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.POST_NOTIFICATIONS) ==
+            PackageManager.PERMISSION_GRANTED
+        ) {
+            // FCM SDK (and your app) can post notifications.
+        } else if (shouldShowRequestPermissionRationale(Manifest.permission.POST_NOTIFICATIONS)) {
+            // TODO: display an educational UI explaining to the user the features that will be enabled
+            //       by them granting the POST_NOTIFICATION permission. This UI should provide the user
```
Comment (egilmorez) [inline messaging/app/src/main/java/com/google/firebase/example/messaging/kotlin/MainActivity.kt:108]: Same comment as the Android version here regarding commas.

File: messaging/app/src/main/java/com/google/firebase/example/messaging/kotlin/MainActivity.kt [hunk 4]
```diff
@@ -90,4 +96,40 @@ class MainActivity : AppCompatActivity() {
...
+    // [START handle_ask_post_notifications_request]
+    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+        when (requestCode) {
+            NOTIFICATION_REQUEST_CODE -> {
+                // If request is cancelled, the result arrays are empty.
+                if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    // FCM SDK (and your app) can post notifications.
+                } else {
+                    // TODO: Inform user that that your app will not show notifications
```
Comment (egilmorez) [inline messaging/app/src/main/java/com/google/firebase/example/messaging/kotlin/MainActivity.kt:127]: Suggest period at end of this comment.

**Review by egilmorez — APPROVED** [review]
Looks great; same nits for Kotlin as for Java.  Thanks!

File: messaging/app/src/main/java/com/google/firebase/example/messaging/MainActivity.java (hunk 1 above)
Comment (kroikie) [reply messaging/app/src/main/java/com/google/firebase/example/messaging/MainActivity.java:122]: Done!

File: messaging/app/src/main/java/com/google/firebase/example/messaging/MainActivity.java (hunk 2 above)
Comment (kroikie) [reply messaging/app/src/main/java/com/google/firebase/example/messaging/MainActivity.java:143]: Done!

File: messaging/app/src/main/java/com/google/firebase/example/messaging/kotlin/MainActivity.kt (hunk 3 above)
Comment (kroikie) [reply messaging/app/src/main/java/com/google/firebase/example/messaging/kotlin/MainActivity.kt:108]: Done!

File: messaging/app/src/main/java/com/google/firebase/example/messaging/kotlin/MainActivity.kt (hunk 4 above)
Comment (kroikie) [reply messaging/app/src/main/java/com/google/firebase/example/messaging/kotlin/MainActivity.kt:127]: Done!

**Pull request merged**